src/libraries/System.Private.CoreLib/src/System/ThrowHelper.cs

```diff
@@ -599,6 +601,25 @@ internal static void ThrowFormatIndexOutOfRange()
             throw new FormatException(SR.Format_IndexOutOfRange);
         }
 
+        internal static AmbiguousMatchException GetAmbiguousMatchException(MemberInfo memberInfo)
+        {
+            Type? declaringType = memberInfo.DeclaringType;
+            return (declaringType is not null)
+                ? new AmbiguousMatchException(SR.Format(SR.Arg_AmbiguousMatchException_Detailed, declaringType, memberInfo))
+                : new AmbiguousMatchException(SR.Format(SR.Arg_AmbiguousMatchException_Simple, memberInfo));
+        }
+
+        internal static AmbiguousMatchException GetAmbiguousMatchException(Attribute attribute)
+        {
+            return new AmbiguousMatchException(SR.Format(SR.Arg_AmbiguousMatchException_Attribute, attribute));
+        }
+
+        internal static AmbiguousMatchException GetAmbiguousMatchException(CustomAttributeData customAttributeData)
+        {
+            Type attributeType = customAttributeData.GetType();
+            return new AmbiguousMatchException(SR.Format(SR.Arg_AmbiguousMatchException_Detailed, attributeType, customAttributeData));
+        }
+
```

Comment:
Is this used?
```suggestion
```

Reply:
Yes, it's in [src\coreclr\nativeaot\System.Private.CoreLib\src\System\Reflection\Attribute.NativeAot.cs](https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Attribute.NativeAot.cs#L124)

Reply:
OK, then I don't think we need to pass the `customAttributeData.GetType()` into the message. A sample test for `customAttributeData.GetType().ToString()` produces `System.Reflection.RuntimeCustomAttributeData` which I believe will be the same in most attributes if not all.